logstreams/src/test/java/io/zeebe/logstreams/storage/atomix/AtomixLogStorageReaderTest.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.storage.atomix;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.protocols.raft.storage.log.entry.ConfigurationEntry;
+import io.atomix.protocols.raft.zeebe.ZeebeEntry;
+import io.atomix.protocols.raft.zeebe.ZeebeLogAppender.AppendListener;
+import io.atomix.storage.journal.Indexed;
+import io.zeebe.logstreams.spi.LogStorage;
+import io.zeebe.logstreams.util.AtomixLogStorageRule;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.IntStream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AtomixLogStorageReaderTest {
+  private static final ByteBuffer DATA = ByteBuffer.allocate(4).putInt(0, 1);
+  private static final Logger LOGGER = LoggerFactory.getLogger(AtomixLogStorageReaderTest.class);
+
+  private final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private final AtomixLogStorageRule storageRule = new AtomixLogStorageRule(temporaryFolder);
+
+  @Rule public final RuleChain chain = RuleChain.outerRule(temporaryFolder).around(storageRule);
+
+  @Test
+  public void shouldLookUpAddress() {
+    // given
+    final var reader = storageRule.get().newReader();
+    final var entries = List.of(append(1, 4, DATA), append(5, 8, DATA), append(9, 10, DATA));
+
+    // when - then
+    IntStream.range(1, 4)
+        .forEach(
+            i -> assertThat(reader.lookUpApproximateAddress(i)).isEqualTo(entries.get(0).index()));
+    IntStream.range(5, 8)
+        .forEach(
+            i -> assertThat(reader.lookUpApproximateAddress(i)).isEqualTo(entries.get(1).index()));
+  }
+
+  @Test
+  public void shouldReturnInvalidOnEmptyLookUp() {
+    // given
+    final var reader = storageRule.get().newReader();
+
+    // when - then
+    assertThat(reader.lookUpApproximateAddress(1)).isEqualTo(LogStorage.OP_RESULT_INVALID_ADDR);
```

Comment:
please try no to combine when then blocks.

These blocks are to improve readability and separate concerns

Reply:
You mean:

```java
@Test
  public void shouldReturnInvalidOnEmptyLookUp() {
    // given
    final var reader = storageRule.get().newReader();

    // when
    final var address = reader.lookUpApproximateAddress(1);
    
    // then
    assertThat(address).isEqualTo(LogStorage.OP_RESULT_INVALID_ADDR);
}
```

Sure, but then we should consistently do that ;)

Reply:
Yes this looks better. Correct we should, if you see this in your reviews please make a remark

Reply:
I think it is now more clearer or for me it is easier to understand what you are testing thanks for the changes